src/main/webapp/js/components/fields/field-config-modal.jsx

```diff
@@ -0,0 +1,286 @@
+  constructor() {
+    super();
+
+    this.state = {
+      modalIsOpen: false,
+      base: false,
+      editable: false,
+      filterable: false,
+      required: false,
+      fieldType: '',
+    };
+
+    this.afterOpenModal = this.afterOpenModal.bind(this);
+    this.displayFields = this.displayFields.bind(this);
+    this.mapConfigToState = this.mapConfigToState.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.mapStateToConfig = this.mapStateToConfig.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { item } = nextProps;
+    this.mapConfigToState(item);
+  }
+
+  onSubmit() {
+    this.mapStateToConfig();
+    this.props.toggleModal();
+  }
+
+  mapStateToConfig() {
+    const {
+      base, editable, filterable, required, fieldType, entity,
+      displayName, name, relatedEntity, relatedField, format,
+    } = this.state;
+    this.props.item.base = base;
+    this.props.item.editable = editable;
+    this.props.item.filterable = filterable;
+    this.props.item.required = required;
+    this.props.item.fieldType = fieldType;
+    this.props.item.entity = entity;
+    this.props.item.displayName = displayName;
+    this.props.item.name = name;
+    this.props.item.relatedEntity = relatedEntity;
+    this.props.item.relatedField = relatedField;
+    this.props.item.format = format;
+  }
+
+  afterOpenModal() {
+    // references are now sync'd and can be accessed.
+    this.subtitle.style.color = '#f00';
+  }
+
+  drawCheckbox(displayName, name, val) {
+    return (
+      <span className="input-row">
+        <span className="field-config-label">{displayName}</span>
+        <Checkbox
+          onChange={value => this.handleChange(value, name)}
+          value={val}
+          className="field-config-checkbox"
+        />
+      </span>
+    );
+  }
+
+  drawSelect(displayName, name, val, options) {
+    return (
+      <span className="input-row">
+        <span className="field-config-label">{displayName}</span>
+        <Select
+          onChange={value => this.handleChange(value, name)}
+          options={FIELD_TYPE_OPTIONS}
+          defaultValue={getDefaultSelectOption(options, val)}
+        />
+      </span>
+    );
+  }
+
+  drawInput(displayName, name, val) {
+    return (
+      <span className="input-row">
+        <span className="field-config-label">{displayName}</span>
+        <Input
+          onChange={value => this.handleChange(value, name)}
+          value={val || ''}
+        />
+      </span>
+    );
+  }
+
+  displayFields(item) {
+    const {
+      base, editable, filterable, required, fieldType, entity,
+      displayName, name, format, relatedEntity, relatedField,
+    } = this.state;
+    return (
+      !!item && (
+      <div>
+        <Form
+          onSubmit={this.onSubmit}
+          render={({ handleSubmit }) => (
+            <form onSubmit={handleSubmit} className="modal-fields">
```

Comment:
I think you can use form utils for this, you just need to create a config in json and map it using the utils method